File: app/src/main/java/life/plank/juna/zone/view/adapter/BoardMembersViewAdapter.java
package life.plank.juna.zone.view.adapter;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import life.plank.juna.zone.R;
import life.plank.juna.zone.data.model.User;
import life.plank.juna.zone.view.activity.InviteToBoardActivity;
import life.plank.juna.zone.view.fragment.board.user.PrivateBoardInfoFragment;

import static life.plank.juna.zone.util.UIDisplayUtil.loadBitmap;

public class BoardMembersViewAdapter extends RecyclerView.Adapter<BoardMembersViewAdapter.BoardMembersViewHolder> {

    public static Bitmap parentViewBitmap = null;
    private Picasso picasso;
    private List<User> userList;
    private Context context;
    private String boardId;
    private String displayName;
    private String boardName;
    private PrivateBoardInfoFragment fragment;

    public BoardMembersViewAdapter(List<User> userList, Context context, String boardId, PrivateBoardInfoFragment fragment, String displayName,
                                   Picasso picasso, String boardName) {
        this.context = context;
        this.userList = userList;
        this.boardId = boardId;
        this.fragment = fragment;
        this.displayName = displayName;
        this.picasso = picasso;
        this.boardName = boardName;
    }

    @Override
    public BoardMembersViewAdapter.BoardMembersViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new BoardMembersViewAdapter.BoardMembersViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user, parent, false));
    }

    @Override
    public void onBindViewHolder(BoardMembersViewAdapter.BoardMembersViewHolder holder, int position) {
        holder.usernameTextView.setText(userList.get(position).getDisplayName());

        picasso.load(userList.get(position).getProfilePictureUrl())
                .into(holder.profileImageView);

        if (userList.get(position).getDisplayName().equals(context.getString(R.string.invite_string))) {
            holder.profileImageView.setBackground(context.getDrawable(R.drawable.new_board_circle));
        }

        holder.profileImageView.setOnClickListener(view -> {
            if (userList.get(position).getDisplayName().equals(context.getString(R.string.invite_string))) {
                parentViewBitmap = loadBitmap(fragment.getActivity().getWindow().getDecorView(), fragment.getActivity().getWindow().getDecorView(), context);
                Intent inviteToBoard = new Intent(context, InviteToBoardActivity.class);
                inviteToBoard.putExtra(context.getString(R.string.intent_board_id), boardId);
                inviteToBoard.putExtra(context.getString(R.string.board_title), context.getString(R.string.invite_people_to) + " " + boardName + " " +
                        context.getString(R.string.board));
                inviteToBoard.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                context.startActivity(inviteToBoard);
            } else if (!userList.get(position).getDisplayName().equals(displayName)) {
                PrivateBoardInfoFragment.onClickProfileImage(view, userList.get(position).getObjectId(), position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return userList.size();
    }

    public void update(List<User> users) {
        userList.clear();
        this.userList.addAll(users);
        notifyDataSetChanged();
    }

    class BoardMembersViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.image)
        ImageView profileImageView;
        @BindView(R.id.title)
        TextView usernameTextView;

        BoardMembersViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/adapter/BoardMembersViewAdapter.kt
package life.plank.juna.zone.view.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_user.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.ZoneApplication
import life.plank.juna.zone.data.model.User
import life.plank.juna.zone.util.DataUtil.findString
import life.plank.juna.zone.util.PreferenceManager
import life.plank.juna.zone.util.customview.CustomPopup
import life.plank.juna.zone.view.activity.InviteToBoardActivity
import life.plank.juna.zone.view.fragment.base.BaseFragment
import life.plank.juna.zone.view.fragment.board.user.PrivateBoardInfoFragment
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.newTask
import org.jetbrains.anko.sdk27.coroutines.onClick

class BoardMembersViewAdapter(
        private var userList: MutableList<User>,
        private val boardId: String,
        private val fragment: BaseFragment,
        private val displayName: String,
        private val boardName: String
) : RecyclerView.Adapter<BoardMembersViewAdapter.BoardMembersViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BoardMembersViewAdapter.BoardMembersViewHolder =
            BoardMembersViewAdapter.BoardMembersViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_user, parent, false))

    override fun onBindViewHolder(holder: BoardMembersViewAdapter.BoardMembersViewHolder, position: Int) {
        holder.itemView.title.text = userList[position].displayName

        Glide.with(fragment)
                .load(userList[position].profilePictureUrl)
                .into(holder.itemView.image)

        if (userList[position].displayName == findString(R.string.invite_string)) {
            holder.itemView.image.background = ZoneApplication.getContext().getDrawable(R.drawable.new_board_circle)
        }

        holder.itemView.image.onClick {
            if (userList[position].displayName == findString(R.string.invite_string)) {
                showInviteToBoardPopup()
            } else if (userList[position].displayName != displayName) {
                showOptionsPopup(holder, position)
            }
        }
    }

    private fun showInviteToBoardPopup() {
        fragment.activity?.run {
            startActivity(intentFor<InviteToBoardActivity>(
                    findString(R.string.intent_board_id) to boardId,
                    findString(R.string.board_title) to (findString(R.string.invite_people_to) + " " + boardName + " " + findString(R.string.board))
            ).newTask())
        }
    }

    private fun showOptionsPopup(holder: BoardMembersViewHolder, position: Int) {
        if (PreferenceManager.getSharedPrefs(findString(R.string.pref_user_details))
                        .getString(ZoneApplication.getContext().getString(R.string.pref_display_name), "NA") == displayName) {
            if (fragment is PrivateBoardInfoFragment) {
                fragment.userPosition = position
                CustomPopup.showPrivateBoardOptionPopup(holder.itemView.image, fragment, userList[position].objectId!!)
            }
        }
    }

    override fun getItemCount(): Int = userList.size

    fun update(users: MutableList<User>) {
        userList = users
        notifyDataSetChanged()
    }

    class BoardMembersViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
